### PR TITLE
fix: use auto height on chart-wrapper unless autosize fit set

### PR DESCRIPTION
**Pull request by metasoarous**
addresses part of #584

### CODE REVIEW

**Comment by metasoarous** [comment]
@domoritz Didn't know about `?.`; Thanks for the cleanup.

**Comment by domoritz** [comment]
It's a new feature in JS and pretty useful, I think: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Review by haldenl — APPROVED** [review]
Looks good to me. Thank you @metasoarous .

Note: this doesn't cover the case in #584 where auto-sizing is set dynamically by the spec (e.g., through a signal). This was the case I was hung up on and never got around to addressing. Thinking about it again the signal case seems like it would be exceedingly rare and this fix for the static case should have been addressed quickly. Apologies for dropping the ball here. I have been out of touch...

File: src/embed.ts [hunk 1]
```diff
@@ -323,6 +323,14 @@ async function _embed(
     chartWrapper.classList.add(CHART_WRAPPER_CLASS);
     element.appendChild(chartWrapper);
     container = chartWrapper;
+    // apply autosize styles to chartWrapper element
+    let autosize = spec?.autosize.type;
```
[the remark this answers is not in the document]
Comment (domoritz) [reply src/embed.ts:327]: And what do we do if autosize is a signal? Does this mean we should maybe write this code differently and instead add a signal listener for `autosize`?

File: src/embed.ts [hunk 2]
```diff
@@ -323,6 +323,14 @@ async function _embed(
     chartWrapper.classList.add(CHART_WRAPPER_CLASS);
     element.appendChild(chartWrapper);
     container = chartWrapper;
+    // apply autosize styles to chartWrapper element
+    let autosize = spec.autosize && spec.autosize.type;
```
[the remark this answers is not in the document]
Comment (domoritz) [reply src/embed.ts]: Sorry, typo `let autosize = spec.autosize?.type;`

**Review by haldenl — APPROVED** [review]

**Comment by domoritz** [comment]
I added some unit tests in bf1bc1c

**Comment by domoritz** [comment]
Thank you @metasoarous for kicking off the improvement.

**Pull request merged**